var/spack/repos/builtin/packages/fftw/package.py

```diff
@@ -96,6 +96,12 @@ def setup_build_environment(self, env):
                 'CXXFLAGS', self.spec['llvm-openmp'].headers.include_flags)
             env.append_flags(
                 'LDFLAGS', self.spec['llvm-openmp'].libs.ld_flags)
+        # FFTW first configures libtool without MPI, and later uses it with
+        # MPI. libtool then calls wrong linker to create shared libraries
+        # (it calls `$CC` instead of `$MPICC`), and MPI symbols
+        # remain undefined because `-lmpi` is not passed to the linker.
+        if self.spec.satisfies('+mpi'):
```

Comment:
Only a problem on macOS:
```suggestion
        if self.spec.satisfies('+mpi') and self.spec.satisfies('platform=darwin'):
```